bigframes/dataframe.py

```diff
@@ -2530,3 +2547,70 @@ def _get_block(self) -> blocks.Block:
 
     def _cached(self) -> DataFrame:
         return DataFrame(self._block.cached())
+
+    _DataFrameOrSeries = typing.TypeVar("_DataFrameOrSeries")
+
+    def dot(self, other: _DataFrameOrSeries) -> _DataFrameOrSeries:
```

Comment:
Lets support multi-index or at least throw a NotImplementedException otherwise

Reply:
Tried multi-index which introduced one extra join and the 10m scale broke. I'll need more time to tune it, so for now I'll resort to NotImplementedException for multi-index. Sending the change in an hour or so.

```
Resources exceeded during query execution: Your project or organization exceeded the maximum disk and memory limit available for shuffle operations. Consider provisioning more slots, reducing query concurrency, or using more efficient logic in this job.
```

Reply:
Done in the latest commit.